src/classes/ng-uploader-options.class.ts

```diff
@@ -22,44 +24,49 @@ export interface INgUploaderOptions {
 
 export class NgUploaderOptions implements INgUploaderOptions {
   url: string;
-  cors?: boolean;
-  withCredentials?: boolean;
-  multiple?: boolean;
+  cors: boolean;
+  withCredentials: boolean;
+  multiple: boolean;
   maxUploads?: number;
   data?: any;
-  autoUpload?: boolean;
+  autoUpload: boolean;
   multipart?: any;
-  method?: 'POST' | 'GET' | 'PATCH' | 'PUT';
-  customHeaders?: any;
-  encodeHeaders?: boolean;
-  authTokenPrefix?: string;
+  method: Method;
+  customHeaders: any;
+  encodeHeaders: boolean;
+  authTokenPrefix: string;
   authToken?: string;
-  fieldName?: string;
-  fieldReset?: boolean;
+  fieldName: string;
+  fieldReset: boolean;
   previewUrl?: string;
-  calculateSpeed?: boolean;
-  filterExtensions?: boolean;
-  allowedExtensions?: string[];
+  calculateSpeed: boolean;
+  filterExtensions: boolean;
+  allowedExtensions: string[];
 
   constructor(obj: INgUploaderOptions) {
-    this.url = obj.url != null ? obj.url : '';
-    this.cors = obj.cors != null ? obj.cors : true;
-    this.withCredentials = obj.withCredentials != null ? obj.withCredentials : this.withCredentials;
-    this.multiple = obj.multiple != null ? obj.multiple : true;
-    this.maxUploads = obj.maxUploads != null ? obj.maxUploads : 10;
-    this.data = obj.data != null ? obj.data : {};
-    this.autoUpload = obj.autoUpload != null ? obj.autoUpload : true;
-    this.multipart = obj.multipart != null ? obj.multipart : false;
-    this.method = obj.method != null ? obj.method : 'POST';
-    this.customHeaders = obj.customHeaders != null ? obj.customHeaders : { };
-    this.encodeHeaders = obj.encodeHeaders != null ? obj.encodeHeaders : false;
-    this.authTokenPrefix = obj.authTokenPrefix != null ? obj.authTokenPrefix : 'Bearer';
-    this.authToken = obj.authToken != null ? obj.authToken : null;
-    this.fieldName = obj.fieldName != null ? obj.fieldName : 'file';
-    this.fieldReset = obj.fieldReset != null ? obj.fieldReset : null;
-    this.previewUrl = obj.previewUrl != null ? obj.previewUrl : null;
-    this.calculateSpeed = obj.calculateSpeed != null ? obj.calculateSpeed : true;
-    this.filterExtensions = obj.filterExtensions != null ? obj.filterExtensions : false;
-    this.allowedExtensions = obj && obj.allowedExtensions ? obj.allowedExtensions : [];
+    function use<T>(source: T|undefined, defaultValue: T): T {
```

Comment:
really nice idea, but it does not compile in my case
```sh
ng-uploader-options.class.ts:51:20: Argument of type 'string' is not assignable to parameter of type '"" | undefined'.
```
please fix and try to build `npm run build` to see if works.